Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -1486,6 +1486,60 @@ namespace OpenInfraPlatform {
 					}
 				}
 
+
+
+				void convertIfcPlanarExtent(const EXPRESSReference<typename IfcEntityTypesT::IfcPlanarExtent> planarExtent,
+					const carve::math::Matrix& pos,
+					std::shared_ptr<ItemData>& itemData)
+					const noexcept(false)
+				{
+					std::vector<carve::geom::vector<3>> loops;
+					std::vector<carve::geom::vector<3>> segment_start_points;
+					convertIfcPlanarExtent(planarExtent, loops, segment_start_points);
+
+					std::shared_ptr<carve::input::PolylineSetData> polylineData = std::make_shared<carve::input::PolylineSetData>();
+					polylineData->beginPolyline();
+					for (int i = 0; i < loops.size(); ++i) {
+						polylineData->addVertex(pos * loops.at(i));
+						polylineData->addPolylineIndex(i);
+					}
+					itemData->polylines.push_back(polylineData);
+				}
+
+
+				void convertIfcPlanarExtent(const EXPRESSReference<typename IfcEntityTypesT::IfcPlanarExtent> planarExtent,
+					std::vector<carve::geom::vector<3>>& loops,
+					std::vector<carve::geom::vector<3>>& segmentStartPoints)
+					const noexcept(false) 
+				{
+					double 	sizeInX = planarExtent->SizeInX;
+					double 	sizeInY = planarExtent->SizeInY;
+
+					//An extent is a rectangle specified by providing the coordinate of the lower left corner and the coordinate of the upper right corner in map units.
+					carve::geom::vector<3> lowerLeftCorner = carve::geom::VECTOR(-sizeInX, -sizeInY, 0.);
+					carve::geom::vector<3> upperRightCorner = carve::geom::VECTOR(sizeInX, sizeInY, 0.);

```

Comment:
This is wrong. Reading [here](https://standards.buildingsmart.org/IFC/RELEASE/IFC4/FINAL/HTML/schema/ifcpresentationdefinitionresource/lexical/ifcplanarbox.htm): 

> The origin of this local coordinate system serves as the lower left corner of the rectangular box.

So `lowerLeftCorner` should be `(0,0)`.

Reply:
Resolved: b4bb324